ironfish/src/wallet/wallet.ts

```diff
@@ -187,6 +187,7 @@ export class Wallet {
     try {
       let hashChanged = false
       do {
+        this.chainProcessor.hash = await this.getEarliestHeadHash()
```

Comment:
i don't think that iterating over heads will have an impact relative to the current performance since right now we're checking `isAccountUpToDate` on every account before decrypting notes